src/ext/java/org/opentripplanner/ext/transmodelapi/model/plan/LegType.java

```diff
@@ -442,6 +445,19 @@ public static GraphQLObjectType create(
           )
           .build()
       )
+      .field(
+        GraphQLFieldDefinition
+          .newFieldDefinition()
+          .description("The leg's elevation profile.")
+          .name("elevationProfile")
+          .type(new GraphQLNonNull(new GraphQLList(elevationStepType)))
+          .dataFetcher(env ->
+            Objects
+              .requireNonNullElse(leg(env).getElevationProfile(), ElevationProfile.empty())
+              .stepsWithoutUnknowns()
```

Comment:
This mapping can be extracted and moved into the `ElevationProfileStepType`. I know we have mappers, but I find it easier to avoid mistakes if the mapping both ways are in the same class. So, having a type like `ElevationProfileStepType` witch define the mapping both ways is nice.

I did the same thing here: https://github.com/opentripplanner/OpenTripPlanner/blob/89f259840b6734832a86f876d079982f09a23768/src/ext/java/org/opentripplanner/ext/transmodelapi/model/framework/StreetModeDurationInputType.java

Reply:
I've extracted the mapper and used it in both the leg and the step.